Compiler/test/EAUtils.jl

```diff
@@ -2,27 +2,34 @@ module EAUtils
 
 export code_escapes, @code_escapes, __clear_cache!
 
-const CC = Core.Compiler
+if !@isdefined(Compiler)
+    if Base.identify_package("Compiler") === nothing
+        import Base.Compiler: Compiler
+    else
+        import Compiler
+    end
+end
```

Comment:
Shouldn’t this actually check whether `Base.REFLECTION_COMPILER` is set?  
As it stands, if `Compiler.jl` exists in the environment where `test/runtests.jl` is run, there’s a mismatch between the compiler used for reflection and the compiler used for testing, which can result in errors.

Reply:
(though this might be best addressed in #56636, where I made another common utility file `setup.jl`)